Pages/Table.php

```diff
@@ -452,6 +452,15 @@ public function postInsert() {
             $values[$this->singularity] = $this->singularityID;
         }
         $this->id = Database::getInstance()->insert($this->table, $values, $this->update_on_duplicate_key ? $values : true);
+        
+        /*
+         * Check if id is defined. If it's FALSE, there was an error 
+         * inserting the new row. Probably duplicating.
+         */
+        if ($this->id == FALSE) {
+            Output::error('User with such email already exists');
```

Comment:
This is for more than just the user table so the error should be more generic. 'There was a conflict with an existing entry'

Also can we just set the error message and return to the 'insert' form? so that they can change the data and enter the new one? If that gets complicated then forget it and we'll just add this one for now.